File: leetcode/C++/array-nesting.cpp
class Solution {
public:
    int arrayNesting(vector<int>& nums) {
        int sol=0;
        vector<int> visit(nums.size(),0);
        for(int i=0,n=0,t=0; i<nums.size(); t=++i){
            n=0;
            while(visit[t]==0) {
                visit[t]=1;
                t=nums[t];
                ++n;
            }
            sol=max(sol,n);
        }
        return sol;
    }
};

File: leetcode/Python/array-nesting.py
class Solution:
    def arrayNesting(self, nums: List[int]) -> int:
        sol, visit, n, t = 0, [0]*len(nums), 0, 0
        for i in range(len(nums)):
            t, n = i, 0
            while visit[t]==0:
                visit[t], t, n = 1, nums[t], n+1
            sol=max(sol,n)
        return sol